src/representation_theory/basic.lean

```diff
@@ -264,6 +261,13 @@ begin
   simp only [of_mul_action_def, finsupp.lmap_domain_apply, finsupp.map_domain_apply, hg],
 end
 
+lemma of_mul_action_as_module_eq_mul
```

Comment:
I think it is worth adding the following lemma above this one:
```lean
@[simps] def of_mul_action_self_as_module_equiv :
  (of_mul_action k G G).as_module ≃ₗ[k] monoid_algebra k G :=
```

I also have a naming suggestion:
```suggestion
lemma of_mul_action_self_smul_eq_mul
```

Reply:
I've added this as a `k[G]`-module isomorphism, using the lemma `of_mul_action_self_smul_eq_mul`, hoping that's what you mean